src/overlays/actors/ovl_En_Karebaba/z_en_karebaba.c

```diff
@@ -9,67 +13,517 @@ void EnKarebaba_Destroy(Actor* thisx, GlobalContext* globalCtx);
+    this->bodyCollider.body.bumper.flags = ~0x00300000;
+    this->headCollider.dim.height = 25;
+}
+
+void EnKarebaba_SetupGrow(EnKarebaba* this) {
+    Actor_SetScale(&this->actor, 0.0f);
+    this->actor.shape.rot.x = -0x4000;
+    this->actionFunc = EnKarebaba_Grow;
+    this->actor.posRot.pos.y = this->actor.initPosRot.pos.y + 14.0f;
+}
+
+void EnKarebaba_SetupIdle(EnKarebaba* this) {
+    Actor_SetScale(&this->actor, 0.005f);
+    this->actor.shape.rot.x = -0x4000;
+    this->actionFunc = EnKarebaba_Idle;
+    this->actor.posRot.pos.y = this->actor.initPosRot.pos.y + 14.0f;
+}
+
+void EnKarebaba_SetupAwaken(EnKarebaba* this) {
+    AnimationHeader* anim = &D_060002B8;
+
+    SkelAnime_ChangeAnim(&this->skelAnime, anim, 4.0f, 0, SkelAnime_GetFrameCount(anim), 0, -3);
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_DUMMY482);
+    this->actionFunc = EnKarebaba_Awaken;
+}
+
+void EnKarebaba_SetupUpright(EnKarebaba* this) {
+    if (this->actionFunc != EnKarebaba_Spin) {
+        Actor_SetScale(&this->actor, 0.01f);
+        this->bodyCollider.base.type = COLTYPE_UNK6;
+        this->bodyCollider.base.acFlags &= ~0x0004;
+        this->bodyCollider.body.bumper.flags = gSaveContext.linkAge != 0 ? 0x07C00710 : 0x0FC00710;
+        this->bodyCollider.dim.radius = 15;
+        this->bodyCollider.dim.height = 80;
+        this->headCollider.dim.height = 80;
+    }
+
+    this->actor.params = 40;
+    this->actionFunc = EnKarebaba_Upright;
+}
+
+void EnKarebaba_SetupSpin(EnKarebaba* this) {
+    this->actor.params = 40;
+    this->actionFunc = EnKarebaba_Spin;
+}
+
+void EnKarebaba_SetupDying(EnKarebaba* this) {
+    this->actor.params = 0;
+    this->actor.gravity = -0.8f;
+    this->actor.velocity.y = 4.0f;
+    this->actor.posRot.rot.y = this->actor.shape.rot.y + 0x8000;
+    this->actor.speedXZ = 3.0f;
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_DEKU_JR_DEAD);
+    this->actor.flags |= 0x30;
+    this->actionFunc = EnKarebaba_Dying;
+}
+
+void EnKarebaba_SetupDeadItemDrop(EnKarebaba *this, GlobalContext *globalCtx) {
+    Actor_SetScale(&this->actor, 0.03f);
+    this->actor.shape.rot.x -= 0x4000;
+    this->actor.shape.unk_08 = 1000.0f;
+    this->actor.gravity = 0.0f;
+    this->actor.velocity.y = 0.0f;
+    this->actor.shape.unk_10 = 3.0f;
+    Actor_ChangeType(globalCtx, &globalCtx->actorCtx, &this->actor, ACTORTYPE_MISC);
+    this->actor.params = 200;
+    this->actor.flags &= ~0x20;
+    this->actionFunc = EnKarebaba_DeadItemDrop;
+}
+
+void EnKarebaba_SetupRetract(EnKarebaba* this) {
+    AnimationHeader* anim = &D_060002B8;
+
+    SkelAnime_ChangeAnim(&this->skelAnime, anim, -3.0f, SkelAnime_GetFrameCount(anim), 0.0f, 2, -3.0f);
+    EnKarebaba_ResetCollider(this);
+    this->actionFunc = EnKarebaba_Retract;
+}
+
+void EnKarebaba_SetupDead(EnKarebaba* this) {
+    AnimationHeader* anim = &D_060002B8;
+
+    SkelAnime_ChangeAnim(&this->skelAnime, anim, 0.0f, 0.0f, 0.0f, 2, 0.0f); // Anim Mode == STOP
```

Comment:
is the anim temp required? and with the two other `SkelAnime_ChangeAnim` calls?

Reply:
Nope! Fixed.